### PR TITLE
Add Dolphin Sockets Tests to the regression tests

### PR DESCRIPTION
**Pull request by daniels220**
There's not much there, but sure, why not run it. Had to remove a couple assertions about host names--it's hard to come up with a host name that has a stable IP address and reverse-DNS record...

### CODE REVIEW

**Comment by daniels220** [comment]
Huh, I see that `InternetAddressTest>>testLocalHost` failed in the CI build. It works on my machine, which I'm guessing comes down to a difference in configuration. The test should obviously not be that brittle, but I don't even know *how* exactly it's failing in CI. Any thoughts?

**Comment by blairmcg** [comment]
How about google-public-dns-a.google.com?

Regarding the failing test, I would start by adding some descriptions to the assertions in the test, and also replacing any generic assert: expressions with assert:equals:, etc, where appropriate. You might also want to consider setting up an appveyor build on your fork to make it easier to run a bunch of test commits through builds. This is relatively straightforward as the build is described in a checked in yaml file.

File: Core/Object Arts/Dolphin/Sockets/InternetAddressTest.cls [hunk 1]
```diff
@@ -17,19 +17,17 @@ testAllForHost
 	"Microsoft should have more than one IP associated with microsoft.com"
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/Sockets/InternetAddressTest.cls]: I suggest replacing with: 
```smalltalk
	"dns.google has two fixed and well-known IP addresses"
	addresses := InternetAddress allForHost: 'dns.google'.
	self assert: addresses size equals: 2.
	self assert: (addresses collect: [:each | each ipAddress]) asSet
		equals: #(#[8 8 4 4] #[8 8 8 8]) asSet.
```

File: Core/Object Arts/Dolphin/Sockets/InternetAddressTest.cls [hunk 2]
```diff
@@ -17,19 +17,17 @@ testAllForHost
...
 testHost
 	| name ip |
 	self isOnline ifFalse: [^self].
 	name := InternetAddress host: 'uk2.net'.
 	ip := InternetAddress ipAddress: name ipAddress.
-	self assert: ip host = name host!
+	self assert: (ip host endsWith: name host)!
 
 testIsIPString
 	self deny: (InternetAddress isIPString: '').
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/Sockets/InternetAddressTest.cls]: Replacing testLocalHost with something like the below will make it clear which of the assertions is failing:
```smalltalk
testLocalHost
	| local computerName |
	computerName := SessionManager current computerName asLowercase.
	local := InternetAddress localHost.
	self assert: local host asLowercase equals: computerName.
	self assert: ((InternetAddress ipAddress: local ipAddress) host subStrings: $.) first asLowercase
		equals: computerName
```

**Comment by daniels220** [comment]
Good ideas, thanks—trying them out, we'll see how it goes with `testLocalHost`. Is there a problem with just using the automatic AppVeyor build here? If it costs you money or bothers you with emails or something I can set up my own, but otherwise that seems like what it's for...

**Comment by blairmcg** [comment]
No problem at all. The only reason I suggested it was because finding there are errors after raising a PR is a little late. By the time I raise a PR I prefer to know that I'm done apart from responding to code review comments, but please work in whatever way you prefer.

BTW: I don't think it would be unreasonable to add a 'self skipIfCiBuild' to the failing test and raise an issue to track it. It still represents a useful improvement.

**Comment by blairmcg** [comment]
As this has been waiting here for a month and a half, I'm going to merge it even though that will break the build. I'll then add a skip for the test that doesn't run on the CI machine.

**Comment by daniels220** [comment]
Sorry for letting this slip through the cracks—I more or less meant to do that (add a skip) myself, and just kept not getting around to it. Thanks for taking care of it.

**Pull request merged**